app/models/postcode_local_restriction_search.rb

```diff
@@ -0,0 +1,53 @@
+class PostcodeLocalRestrictionSearch
+  UK_POSTCODE_PATTERN = %r{
+    \A
+    ([Gg][Ii][Rr] 0[Aa]{2}) # Unusual postcodes
+    |
+    (
+      # Outward code
+      (([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9][A-Za-z]?))))
+      \s?
+      [0-9][A-Za-z]{2} # Inward code
+    )
+    \Z
+  }x.freeze
+
+  attr_reader :query, :normalised_postcode
+  delegate :no_information?, to: :location_lookup
+
+  def initialize(query)
```

Comment:
I didn't think postcode was the right name for it as 1) we don't know whether the input is a postcode and 2) the formatted version is arguably the definitive postcode.

I can see why this is confusing though, particularly since there is `query` and `normalised_postcode` side by side. I've tried to resolve this by taking a different approach. I've added a method called `postcode` that will returned the formatted postcode only when there are no errors. query is named `input` and `normalised_postcode` is renamed `normalised_input` (and become private). This will mean there is just one `postcode` method and it only returns when there is a valid postcode.